app/src/main/java/com/duckduckgo/app/settings/db/SettingsDataStore.kt

```diff
@@ -91,11 +91,18 @@ interface SettingsDataStore {
     fun clearAppBackgroundTimestamp()
 }
 
-@ContributesBinding(AppScope::class)
+@ContributesBinding(
+    scope = AppScope::class,
+    boundType = SettingsDataStore::class,
+)
+@ContributesBinding(
+    scope = AppScope::class,
+    boundType = AutoCompleteSettings::class,
+)
```

Comment:
```suggestion
@ContributesBinding(
    scope = AppScope::class,
    boundType = SettingsDataStore::class,
)
@ContributesBinding(
    scope = AppScope::class,
    boundType = AutoCompleteSettings::class,
)
```

If `@SingleInstanceIn` is not defined, each binding will create a new instance. So, for example, in the `GeneralSettingsViewModel` you'll now receive two different instances of `SettingsSharedPreferences` under two different interfaces.

I'm not sure if that's problematic. If the fact that `SettingsSharedPreferences` is not a singleton wasn't causing issues until this point, then it's likely fine. In the end, the underlying shared prefs object is a singleton anyway, but just wanted to flag.